ceph/rados/mgr_workflows.py

```diff
@@ -0,0 +1,274 @@
+        self.client.exec_command(sudo=True, cmd=cmd)
+        time.sleep(10)
+
+        endtime = datetime.datetime.now() + datetime.timedelta(seconds=180)
+        while datetime.datetime.now() < endtime:
+            if self.check_mgr_exists_on_host(host=hostname):
+                log.info(f"Mgr {hostname} is add to the cluster.")
+                return True
+            log.info(
+                "Mgr daemon still not added.sleeping for 5 seconds and checking again"
+            )
+            time.sleep(5)
+        return False
+
+    def remove_mgr_service(self, host) -> bool:
+        """
+        Method to remove the mgr service from the given host
+
+        Args:
+            host: name of the host from where mgr needs to be removed
+
+        returns:
+            Pass -> True, Fail -> false
+        """
+        mgr_daemon_list = self.get_mgr_daemon_list()
+        new_mgr_daemon = ""
+        for node in mgr_daemon_list:
+            if host.hostname in node:
+                new_mgr_daemon = node.strip()
+                break
+
+        host_name = host.hostname
+        if not self.check_mgr_exists_on_host(host=host_name):
+            log.info(f"MGR daemon not present on the host {host_name}. Returning Pass")
+            return True
+
+        # setting mgr daemon unmanaged to true
+        unmanaged_cmd_true = " ceph orch apply mgr  --unmanaged=true"
+        self.client.exec_command(sudo=True, cmd=unmanaged_cmd_true)
+
+        cmd = f"ceph orch daemon rm mgr.{new_mgr_daemon} --force"
+        try:
+            self.client.exec_command(sudo=True, cmd=cmd)
+        except Exception as err:
+            log.debug(f"Hit issue during command execution : {err}")
+            log.debug("proceeding to check if the command execution was successful")
+
+        mgr_exists = True
+        endtime = datetime.datetime.now() + datetime.timedelta(seconds=180)
+        while datetime.datetime.now() < endtime:
+            if not self.check_mgr_exists_on_host(host=host_name):
+                log.info(f"Mgr daemon is removed on the host {host_name}.")
+                mgr_exists = False
+                break
+            log.info(
+                f"Mgr daemon still present on the host {host_name}. sleeping for 5 seconds and checking again"
+            )
+            time.sleep(5)
+        if mgr_exists:
+            log.error(
+                f"Mgr daemon still present on the host {host_name}. Failed to remove"
+            )
+            return False
+
+        # Removing mgr label from host if present
+        labels = self.mon_obj.get_host_labels(host=host_name)
+        if "mgr" in labels:
+            cmd = f"ceph orch host label rm {host_name} mgr"
+            self.client.exec_command(sudo=True, cmd=cmd)
+            log.debug(f"Removed mgr label from host : {host_name}")
+            # setting mgr daemon unmanaged to true
+
+        unmanaged_cmd_false = " ceph orch apply mgr  --unmanaged=false"
```

Comment:
Need to update the way unmanaged flag is applied on the cluster. Without that the placement strategy for the daemon would change.

Reply:
I changed the command to manage and unmanaged the daemon.
Command used - ceph orch set-unmanaged mgr and ceph orch set-managed mgr